File: Java/2079_Watering_Plants.java
class Solution {
    public int wateringPlants(int[] plants, int capacity) {
        int steps = 1;
        int temp = capacity;
        
        temp -= plants[0];
        plants[0] = 0;

        for(int i = 0; i < plants.length-1; i++){
            
            if(plants[i+1] > temp){
                steps += 2*(i+1);
                temp = capacity;
            }
            
            steps++;
            temp -= plants[i+1];
            plants[i+1] = 0;
        }
        return steps;
    }
}

File: CS/2079_Watering_Plants.cs
public class Solution {
    public int WateringPlants(int[] plants, int capacity) {
        int result = 0;
        int c = capacity;
        
        for(int i = 0; i < plants.Length; i++){
            
            result++;
            c -= plants[i];
            plants[i] = 0;
            
            if(i != plants.Length -1){
                if(c < plants[i+1]){
                    result += i+2;
                    c = capacity;
                    i = 0;
                }
            }
        }
        return result;
    }
}
